File: Baekjoon/구현/1747.py
# https://www.acmicpc.net/problem/1747

import sys
input = sys.stdin.readline
    
def isPrime(n):
    if n == 1:
        return False
    for i in range(2, int(n ** 0.5)+1):
        if n % i == 0:
            return False
    return True

def isPalindrome(n):
    n = str(n)
    return n == n[::-1]

n = int(input())

while True:
    if isPalindrome(n) and isPrime(n):
        print(n)
        break
    n += 1

File: Baekjoon/구현/1747.cpp
// https://www.acmicpc.net/problem/1747

#include <iostream>
#include <vector>
#include <algorithm>
#include <cmath>
using namespace std;

bool isPrime(int n) {
    if (n < 2) {
        return false;
    }

    for (int i = 2; i <= sqrt(n); i++) {
        if (n % i == 0) {
            return false;
        }
    }

    return true;
}

int main() {
    int n;
    cin >> n;

    for (int i = n; ; i++) {
        if (isPrime(i)) {
            string s = to_string(i);
            string reversedS = s;
            reverse(reversedS.begin(), reversedS.end());

            if (s == reversedS) {
                cout << i << endl;
                break;
            }
        }
    }

    return 0;
}